### PR TITLE
optee-os: Fix imx8 failure after reboot many times

### PR DESCRIPTION
**Pull request by niej**
Update optee-os to fix imx8 failure after reboot many times.

Memory content from previous boot should be cleaned so that dtb overlay will not be added to top of previous dtb overlay again and again.

### CODE REVIEW

**Comment by niej** [comment]
test report:
http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=4&custom_version=jun-imx8-reboot-failure&lava_query=&lava_owner=&failures=false&submitter=junnie01&sort=by_job

File: meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+--- a/core/arch/arm/kernel/generic_entry_a64.S
++++ b/core/arch/arm/kernel/generic_entry_a64.S
+@@ -56,6 +56,10 @@ FUNC _start , :
+ 	mov	x19, x0		/* Save pagable part address */
+ #if defined(CFG_DT_ADDR)
+ 	ldr     x20, =CFG_DT_ADDR
++#ifndef CFG_DT_OVERLAY
++	mov	x14, #0x0       /* clear dtb magic value */
++	str	x14, [x20]
++#endif
```
Comment (bryanodonoghue) [inline meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: This confuses me.
Aren't we using DTB overlay on i.MX8 ?

Comment (niej) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: We are using DTB overlay with CFG_EXTERNAL_DTB_OVERLAY, but not CFG_DT_OVERLAY.

Comment (grandpaul) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: I also got questions to this. As you modified a64 code. But we also got the same problems on arm-32 platforms. How do we address it?

Comment (niej) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: Because there is no suitable configuration to limit the code and the logic is politics as you mention. Also because this PR is a fix to corner case of test system, so I assume this is a temp solution to imx8mm.
Anyway, I already update patch with change in init_overlay code, although the condition on CFG_DT_ADDR still breaks current optee document.

Comment (bryanodonoghue) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: Lets just put the patch into our Linaro tree.

Comment (bryanodonoghue) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: That way we can go back and look at this again if we have to, or propose it for upstream.

Comment (grandpaul) [reply meta-linaro-mbl/meta-optee/recipes-security/optee/optee-os/0002-core-clear-original-dtb-data-in-special-case.patch]: ok. Agree. Let's just keep this patch locally.

**Comment by niej** [comment]
jenkins build and lava test:

http://jenkins.mbed-linux.arm.com/view/JunNie/job/jun-imx8-reboot-failure/5/

http://e120856-lin.cambridge.arm.com/results.php?image_version=custom&image_number=5&custom_version=jun-imx8-reboot-failure&lava_query=&lava_owner=&failures=false&submitter=junnie01&sort=by_job

**Comment by grandpaul** [comment]
I do test the jenkins build jun-imx8-reboot-failure/7/.
It does boot over 100 times.
I'm using the following scripts.


/home/root/reboot_test.sh
```
#!/bin/sh

CONFIG_FILE=/home/root/reboot_test.conf

if [[ -f $CONFIG_FILE ]]; then
    . $CONFIG_FILE
else
    COUNTER=0
fi


COUNTER=`expr $COUNTER + 1`
echo "COUNTER="$COUNTER > $CONFIG_FILE

sleep 60
shutdown -r 1
sleep 1000
```

/lib/systemd/system/reboot_test.service
```
[Unit]
Description=Reboot test

[Service]
ExecStart=/home/root/reboot_test.sh

[Install]
WantedBy=multi-user.target
```

**Comment by bryanodonoghue** [comment]
Removes the previous bug I could trigger with holding down return on the keyboard somehow so merged.
